GVRf/Framework/framework/src/main/java/org/gearvrf/script/GVRScriptBehavior.java

```diff
@@ -0,0 +1,336 @@
+package org.gearvrf.script;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.gearvrf.GVRAndroidResource;
+import org.gearvrf.GVRBehavior;
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRResourceVolume;
+import org.gearvrf.GVRScene;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.IPickEvents;
+import org.gearvrf.ISceneEvents;
+import org.gearvrf.utility.FileNameUtils;
+import org.gearvrf.GVRPicker;
+import org.gearvrf.IPickEvents;
+import org.gearvrf.ISensorEvents;
+import org.gearvrf.SensorEvent;
+
+/**
+ * Attaches a Java or Lua script to a scene object.
+ * 
+ * These script callbacks are invoked if they are present:
+ *      onEarlyInit(GVRContext) called after script is loaded
+ *      onAfterInit()           called when the script becomes active
+ *                              (this component is attached to a scene object and enabled)
+ *      onStep()                called every frame if this component is enabled
+ *                              and attached to a scene object
+ *      onPickEnter(GVRSceneObject, GVRPicker.GVRPickedObject)
+ *                              called when picking ray enters an object
+ *      onPickExit(GVRSceneObject)
+ *                              called when picking ray exits an object
+ *      onPickInside(GVRSceneObject, GVRPicker.GVRPickedObject)
+ *                              called when picking ray is inside an object
+ *      onPick(GVRPicker)       called when picking selection changes
+ *      onNoPick(GVRPicker)     called when nothing is picked
+ *
+ */
+public class GVRScriptBehavior extends GVRBehavior implements IPickEvents, ISensorEvents, ISceneEvents
+{
+    static private long TYPE_SCRIPT_BEHAVIOR = (System.currentTimeMillis() & 0xfffffff);
```

Comment:
I fixed this by adding newComponentType which calculates a type from the component class hashcode and the time.